mm2src/coins/eth/eth_swap_v2/eth_taker_swap_v2.rs

```diff
@@ -766,3 +817,23 @@ fn validate_erc20_taker_payment_data(
     }
     Ok(())
 }
+
+fn get_dex_fee_and_amount_from_eth_payment_data(
+    decoded: &Vec<Token>,
+    tx_value: U256,
+) -> Result<(U256, U256), PrepareTxDataError> {
+    check_decoded_length(decoded, 7)?;
+    let dex_fee = match decoded.get(1) {
+        Some(Token::Uint(dex_fee)) => *dex_fee,
+        _ => {
+            return Err(PrepareTxDataError::Internal(format!(
+                "Invalid token type for dex fee, got decoded function data: {:?}",
+                decoded
+            )))
+        },
+    };
+    let amount = tx_value
+        .checked_sub(dex_fee)
+        .ok_or_else(|| PrepareTxDataError::Internal("Underflow occurred while calculating amount".into()))?;
+    Ok((dex_fee, amount))
+}
```

Comment:
nit: this file is slowly following the path of its bigger brother `eth.rs` with these global functions being added :)

Reply:
this is a priv function which is used only in current file. 
Right now common structs and funcs are stored in `/eth_swap_v2/mod.rs`
as for `eth_taker_swap_v2.rs` and `eth_maker_swap_v2.rs` they are almost done. 
we can do crates refactoring during project refactor.

Reply:
aha. tbh i didn't pay much attention to the structuring. but i wanted to bring attention to this as it always feels off when we have global functions (outside utilities/commons file or something).
this is a nit anyway as it's a task for the new year :)